### PR TITLE
raidboss: ARR dungeons triggers hard

### PR DESCRIPTION
**Pull request by Aho-Senpai**

### CODE REVIEW

File: ui/raidboss/data/02-arr/dungeon/hataliti_hard.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  triggers: [
+    {
+      id: 'Hataliti Hard Demon Eye',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '833', source: 'Catoblepas', capture: false }),
+      alertText: (_data, _matches, output) => output.text!(),
+      outputStrings: {
+        text: {
+          en: 'Use the Orb',
+          fr: 'Utilisez l\'Orb',
```
Comment (mooondark) [inline ui/raidboss/data/02-arr/dungeon/hataliti_hard.ts:19]: fr: 'Utilisez l\'Orbe',

File: ui/raidboss/data/02-arr/dungeon/hataliti_hard.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    },
+    {
+      id: 'Hataliti Hard Standstill',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '84F', source: 'Mumuepo the Beholden', capture: false }),
+      alertText: (_data, _matches, output) => output.text!(),
+      outputStrings: {
+        text: {
+          en: 'Use the Nail',
+          fr: 'Utilisez L\'Epine',
```
Comment (mooondark) [inline ui/raidboss/data/02-arr/dungeon/hataliti_hard.ts:31]: fr: 'Utilisez L\'Épine',

File: ui/raidboss/data/02-arr/dungeon/the_wanderers_palace_hard.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    {
+      // Not 100% sure if there's a better way to handle the callout
+      id: 'Wanderer\'s Palace Hard Doom',
+      type: 'GainsEffect',
+      netRegex: NetRegexes.gainsEffect({ effectId: 'D2' }),
+      alertText: (data, matches, output) => output.text!({ player: data.ShortName(matches.target) }),
+      outputStrings: {
+        text: {
+          en: 'Heal ${player} to full : Doom',
+          fr: 'Soin Complet dur ${player} : Doom',
```
Comment (mooondark) [inline ui/raidboss/data/02-arr/dungeon/the_wanderers_palace_hard.ts:27]: fr: 'Soin complet sur ${player} : Glas',

File: ui/raidboss/data/02-arr/dungeon/the_tam_tara_depocraft_hard.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  triggers: [
+    {
+      id: 'Tam-Tara Hard Inhumanity',
+      type: 'Ability',
+      netRegex: NetRegexes.ability({ id: '956', source: 'Liavinne', capture: false }),
+      alertText: (_data, _matches, output) => output.text!(),
+      outputStrings: {
+        text: {
+          en: 'DO NOT KILL ADDS! Stack up.',
+          fr: 'Packez-vous: ne tuez pas les Adds',
```
Comment (quisquous) [inline ui/raidboss/data/02-arr/dungeon/the_tam_tara_depocraft_hard.ts:19]: The capital letters seem a little extreme here.  What about:

```suggestion
          en: 'Stack (ignore adds)',
```

**Review by quisquous — COMMENTED** [review]
Sorry for the belated PR review here.  I got distracted by ultimate and didn't see that you'd fixed the test/lint errors from before.

File: ui/raidboss/data/02-arr/dungeon/sastasha_hard.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+          cn: '驱散: ${player}',
+          ko: '"${player}" 에스나',
+        },
+      },
+    },
+    {
+      id: 'Sastasha Hard Tail Screw',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: 'BF4', source: 'Karlabos' }),
+      response: Responses.tankBuster(),
```
[the remark this answers is not in the document]
Comment (MaikoTan) [reply ui/raidboss/data/02-arr/dungeon/sastasha_hard.ts]: I think not only "stun", but also notify "Tail Screw on ${player}", so then healer would deal with it if stun failed.

[the remark this answers is not in the document]
Comment (quisquous) [reply ui/raidboss/data/02-arr/dungeon/sastasha_hard.ts]: Hmm, that's a pretty long output, and not everybody can stun.  What about a combination of alert and info instead?  Something like?

```typescript
      netRegex: NetRegexes.startsUsing({ id: 'BF4', source: 'Karlabos' }),
      alertText: (data, matches, output) => {
        if (data.CanStun())
          return output.stun!({name: matches.source});
      },
      infoText: (data, matches, output) => {
        return output.tailScrewOn!({player: data.ShortName(matches.target)});
      },
      outputStrings: {
        stun: Outputs.stun,
        tailScrew: {
          en: 'Tail Screw on ${player}',
        },
      },
    },
```

[the remark this answers is not in the document]
Comment (MaikoTan) [reply ui/raidboss/data/02-arr/dungeon/sastasha_hard.ts]: looks good 👍

**Comment by Aho-Senpai** [comment]
`ReferenceError: outputs is not defined` ? i'm confused

File: ui/raidboss/data/02-arr/dungeon/sastasha_hard.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
+import NetRegexes from '../../../../../resources/netregexes';
+import outputs from '../../../../../resources/outputs';
```
Comment (quisquous) [inline ui/raidboss/data/02-arr/dungeon/sastasha_hard.ts:2]: this should be `import Outputs from ... etc`

**Pull request merged**